cluster_pack/packaging.py

```diff
@@ -0,0 +1,509 @@
+import getpass
+import imp
+import json
+import logging
+import os
+import pathlib
+import shutil
+import subprocess
+from subprocess import Popen, CalledProcessError, PIPE
+import sys
+import tempfile
+from typing import (
+    Optional,
+    Tuple,
+    Dict,
+    NamedTuple,
+    Callable,
+    Collection,
+    List
+)
+from urllib import parse, request
+import uuid
+import zipfile
+import tensorflow as tf
+import __main__
+
+try:
+    import conda_pack
+except NotImplementedError:
+    # conda is not supported on windows
+    pass
+from pex.fetcher import Fetcher, PyPIFetcher
+from pex.pex_builder import PEXBuilder
+from pex.resolvable import Resolvable
+from pex.resolver import resolve_multi, Unsatisfiable, Untranslateable
+from pex.resolver_options import ResolverOptionsBuilder
+from pex.pex_info import PexInfo
+from pex.interpreter import PythonInterpreter
+
+
+CRITEO_PYPI_URL = "http://build-nexus.prod.crto.in/repository/pypi/simple"

```

Comment:
It is very criteo specific, should we do something like CUSTOM_PYPI_URL and add an environment variable >

Reply:
Yes, we should the problem is how to set it. 
- If we set it in consuming repos it would make the api more complicated.
- If we set it on all Critero environments  it would take a long time (all chef recipes must be updated on sre side
- One alternative could be to read this from .pip/pip.conf, and parse the file, everybody using criteo packages should have the settings

Can we do it in another review ? It is already here actually:
https://github.com/criteo/tf-yarn/blob/master/tf_yarn/packaging.py#L42

Reply:
Fine for me in another review